larray/inout/xw_reporting.py

```diff
@@ -0,0 +1,287 @@
+            return row + 2
+
+    _default_items_size['title'] = (1000, 50)
+
+    # TODO: fix problem with nans values
+    class GraphItem(AbstractExcelItem):
+        def __init__(self, title, data, template, top, left, width, height):
+            AbstractExcelItem.__init__(self, top, left, width, height)
+            self.title = str(title)
+            data = aslarray(data)
+            if not 0 < data.ndim <= 2:
+                raise ValueError("Expected 1D or 2D array for data argument. "
+                                 "Got array of dimensions {}".format(data.ndim))
+            self.data = data
+            if template is not None and not os.path.isfile(template):
+                raise ValueError("Could not find template file {}".format(template))
+            self.template = template
+
+        def dump(self, sheet, data_sheet, row):
+            data_range = data_sheet.range
+            data_cells = data_sheet.cells
+            # write graph title in data sheet
+            data_cells(row, 1).value = self.title
+            row += 1
+            # dump data to make the graph in data sheet
+            data = self.data
+            nb_series = 1 if data.ndim == 1 else data.shape[0]
+            nb_xticks = data.size if data.ndim == 1 else data.shape[1]
+            last_row, last_col = row + nb_series, nb_xticks + 1
+            data_range(data_cells(row, 1), data_cells(last_row, last_col)).value = data.dump()
+            data_cells(row, 1).value = ''
+            # generate graph in destination sheet
+            sheet_charts = sheet.charts
+            obj_chart = sheet_charts.add(self.left, self.top, self.width, self.height)
+            source = data_range(data_cells(row, 1), data_cells(last_row, last_col))
+            obj_chart.set_source_data(source)
+            native_obj_chart = obj_chart.api[0].Chart
+            native_obj_chart.ChartType = ChartType.xlLine
+            native_obj_chart.HasTitle = True
+            native_obj_chart.ChartTitle.Caption = self.title
+            native_obj_chart.Legend.Position = LegendPosition.xlLegendPositionBottom
+            if self.template is not None:
+                native_obj_chart.ApplyChartTemplate(self.template)
+            # flagflip
+            if nb_series > 1 and nb_xticks == 1:
+                native_obj_chart.PlotBy = RowCol.xlRows
+            # update and return current row position
+            return row + nb_series + 2
+
+    _default_items_size['graph'] = (427, 230)
+
+    class ExcelReport(AbstractExcelReport):
+        def __init__(self, template_dir=None):
+            self.items = OrderedDict()
+            self.cursheet = None
+            self.top = None
+            self.left = None
+            self.curline_height = None
+            self.template_dir = None
+            if template_dir is not None:
+                self.set_template_dir(template_dir)
+            self.template = None
+            self.default_items_size = _default_items_size.copy()
```

Comment:
The way to handle default items size bothers me a bit. Wouldn't class attributes on each AbstractExcelItem class and using the classes themselves as keys (instead of strings) in *_items_size make things cleaner?